diff --git a/node_modules/react-element-to-jsx-string/dist/cjs/index.js b/node_modules/react-element-to-jsx-string/dist/cjs/index.js
index aeeaf0c..cbc0511 100644
--- a/node_modules/react-element-to-jsx-string/dist/cjs/index.js
+++ b/node_modules/react-element-to-jsx-string/dist/cjs/index.js
@@ -290,8 +290,8 @@ function sortObject(value) {
     return value;
   } // return date and regexp values as is
 
-
-  if (value instanceof Date || value instanceof RegExp) {
+  // patch from https://github.com/algolia/react-element-to-jsx-string/pull/619
+  if (value instanceof Date || value instanceof RegExp || React__default.isValidElement(value)) {
     return value;
   } // make a copy of array with each item passed through sortObject()
 
@@ -438,6 +438,10 @@ var formatFunction = (function (fn, options) {
 var formatComplexDataStructure = (function (value, inline, lvl, options) {
   var normalizedValue = sortObject(value);
   var stringifiedValue = dist_1(normalizedValue, {
+    // patch from https://github.com/algolia/react-element-to-jsx-string/pull/620
+    filter: function filter(currentValue) {
+      return !React.isValidElement(currentValue) && typeof currentValue !== 'function';
+    },
     transform: function transform(currentObj, prop, originalResult) {
       var currentValue = currentObj[prop];
 
@@ -450,7 +454,7 @@ var formatComplexDataStructure = (function (value, inline, lvl, options) {
       }
 
       return originalResult;
-    }
+    },
   });
 
   if (inline) {
diff --git a/node_modules/react-element-to-jsx-string/dist/esm/index.js b/node_modules/react-element-to-jsx-string/dist/esm/index.js
index 230eba4..7864926 100644
--- a/node_modules/react-element-to-jsx-string/dist/esm/index.js
+++ b/node_modules/react-element-to-jsx-string/dist/esm/index.js
@@ -283,8 +283,8 @@ function sortObject(value) {
     return value;
   } // return date and regexp values as is
 
-
-  if (value instanceof Date || value instanceof RegExp) {
+  // patch from https://github.com/algolia/react-element-to-jsx-string/pull/619
+  if (value instanceof Date || value instanceof RegExp || isValidElement(value)) {
     return value;
   } // make a copy of array with each item passed through sortObject()
 
@@ -431,6 +431,10 @@ var formatFunction = (function (fn, options) {
 var formatComplexDataStructure = (function (value, inline, lvl, options) {
   var normalizedValue = sortObject(value);
   var stringifiedValue = dist_1(normalizedValue, {
+    // patch from https://github.com/algolia/react-element-to-jsx-string/pull/620
+    filter: function filter(currentValue) {
+      return !isValidElement(currentValue) && typeof currentValue !== 'function';
+    },
     transform: function transform(currentObj, prop, originalResult) {
       var currentValue = currentObj[prop];
 
@@ -443,7 +447,7 @@ var formatComplexDataStructure = (function (value, inline, lvl, options) {
       }
 
       return originalResult;
-    }
+    },
   });
 
   if (inline) {
